modules/core/02-client/keeper/events.go

```diff
@@ -14,8 +14,23 @@ import (
 	"github.com/cosmos/ibc-go/v7/modules/core/exported"
 )
 
-// EmitCreateClientEvent emits a create client event
-func EmitCreateClientEvent(ctx sdk.Context, clientID string, clientState exported.ClientState) {
+// EmitUpgradeChainEvent emits an upgrade chain event.
+func EmitUpgradeChainEvent(ctx sdk.Context, height int64) {
```

Comment:
Is there a reason this is still public?

Reply:
This one I couldn't change because is used [here](https://github.com/cosmos/ibc-go/blob/e093d85b533ab3572b32a7de60b88a0816bed4af/modules/core/02-client/abci.go#L31). But maybe there's also a way around, if we can move the event emission inside `SetUpgradedConsensusState`?

Reply:
I think it might be a good idea to move all our begin blocker actions to the keeper directory (within abci.go). 

```go
func (keeper Keeper) BeginBlocker(ctx sdk.Context) {
    // functionality
}
```
```go
// within ibc module.go
func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
	am.keeper.ClientKeeper.BeginBlocker(ctx)
}